src/OpenTelemetry/Metrics/CircularBufferBuckets.cs

```diff
@@ -267,6 +267,37 @@ static void Move(long[] array, uint src, uint dst)
         }
     }
 
+    internal void Reset()
+    {
+        if (this.trait != null)
+        {
+            for (var i = 0; i < this.trait.Length; ++i)
+            {
+                this.trait[i] = 0;
+            }
+        }
+    }
+
+    internal CircularBufferBuckets Copy()
+    {
+        // TODO: Consider avoiding allocating a new one every time.
+        // Maybe we can have two instances that we swap between snapshots.
```

Comment:
👍

Reply:
Was planning to grab this in a follow up.